MCDataProducts/inc/StrawDigiMC.hh

```diff
@@ -32,16 +32,21 @@ namespace mu2e {
     typedef std::array<SGSP,StrawEnd::nends> SGSPA;
     typedef std::array<XYZVectorF,StrawEnd::nends> PA;
     typedef std::array<float,StrawEnd::nends> FA;
+    enum Validity {Valid, PartiallyValid, Invalid};
```

Comment:
I suggest renaming this enum 'Provenance', with states "sim", "mixed", "external"

Reply:
Should we use an EnumToString so that we have access to printed values?

Reply:
An example of the EnumToString is:
  https://github.com/Mu2e/Offline/blob/main/Mu2eUtilities/inc/TrackPatRecType.hh
 https://github.com/Mu2e/Offline/blob/main/Mu2eUtilities/src/TrackPatRecType.cc

Reply:
@brownd1978 why "external" and not "data"?

@edcallaghan There will be two ways to make pure sim digis.  One is our present way and the other is to make background frames of digis from sim and then to overlay signal sims.  Will be be of Provenance "sim" (in Dave's proposed language?).   If so, should we distinguish them?

Reply:
IMO the "most correct" answer depends on what the conflict resolution looks like for the MC truth data products, which I haven't through through in at a detailed level. If there are no caveats to "merging" the MC truth info for the two constituent sim digis, then labeling the single output digi as "Sim" is fine. But if there is some kind of catch, then we probably want to label is as "Sim/Sim" as opposed to e.g. "Sim" or "Sim/Data".

Reply:
Oh, but also, this is not limited to just two digis input. It won't occur at scale, but in principle arbitrarily many digis can participate. So I'd err on the side of Dave's suggestion of renaming the three situations (purely new MC, mixed new MC with something preexisting, and purely something preexisting). In the case of the second situation, the onus is then on the user to know what it is that they have mixed (preexisting simulation, or real data), which I think is a reasonable assumption if they are looking at MC truth info.

Reply:
I think we should distinguish 3 different kinds of mixed:  (sim+sim), (sim+data) and (data+data).   Maybe this is not the right place to record that information - but we do need a plan for a robust audit trail so that an end user can determine what it is programatically and not by looking at a wiki page or a log book.

An example of data+data would be if we took at track from Offspill and embedded it in a random on-spill trigger as part of a study to understand reconstruction efficiency.

Reply:
I agree that that information (i.e. the mixing constituents/identities) should, for official production, be stored somewhere. I do not have a strong opinion on whether it should be part of the data structure or not. But I do not think that this class is the correct place for that book-keeping. For one thing, if someone is looking at a `StrawDigiMC`, then they are either going down the wrong path to begin with, or they are aware that they have run a simulation --- and because the actual "merging" of data products is done within the `StrawDigisFromStrawGasSteps` simulation module, this PR cannot create a single collection of digis from a "data + data" situation (though, as you say, we may want to accommodate that for future use cases).

Reply:
@brownd1978 @kutschke The latest push exchanges `Validity::{Valid, PartiallyValid, Invalid}` for `Provenance::{unknown, Simulation, Mixed, External}`, which is implemented as an `EnumToStringSparse` specialization. The existence of `unknown` is forced by that interface. In the latter case, the onus is on the user to understand that `External` may either signify that real data or preexisting MC (with or without MC info available yet undetermined, as that will depend on how the collision resolution ends up being performed). But the use case is really just to flag that one should not expect the MC truth object to actually represent MC truth information, as the digi came from an alternate source.

As to resolving the ambiguity of whether it was data or simulation which was provided externally, my current thinking is as follows: the `EventID` of the mixed event is required to be propagated as a data product so that the conditions can be matched. As long as we don't drop that data product from the output, this would satisfy the book-keeping (i.e., the run number of the mixed event tells us an analyst if it came from data or not).

Reply:
Thanks Ed.  We have converged as well as we can for now.  Your suggestion for documenting that data was the external might well work - we can work through the details when the time comes.